config/config/src/getOptionsFromRootManifest.ts

```diff
@@ -44,6 +47,10 @@ export function getOptionsFromRootManifest (manifest: ProjectManifest): {
     // @ts-expect-error
     settings['onlyBuiltDependencies'] = onlyBuiltDependencies
   }
+  if (onlyBuiltDependenciesFile) {
+    // @ts-expect-error
```

Comment:
nitpick: Extracting the interface returned from `getOptionsFromRootManifest` would be a bit more TypeScript idiomatic. It also allows us to remove the `// @ts-expect-error`, which can unexpectedly hide other compile errors unrelated to indexing that happen in the future.

```ts
export interface OptionsFromRootManifest {
  allowedDeprecatedVersions?: AllowedDeprecatedVersions
  allowNonAppliedPatches?: boolean
  overrides?: Record<string, string>
  neverBuiltDependencies?: string[]
  onlyBuiltDependencies?: string[]
  onlyBuiltDependenciesFile?: string
  packageExtensions?: Record<string, PackageExtension>
  patchedDependencies?: Record<string, string>
  peerDependencyRules?: PeerDependencyRules
}

export function getOptionsFromRootManifest (
  manifestDir: string,
  manifest: ProjectManifest
): OptionsFromRootManifest { /* ... */ }
```

And then further down:

```ts
const settings: OptionsFromRootManifest = {
  allowedDeprecatedVersions,
  allowNonAppliedPatches,
  overrides,
  neverBuiltDependencies,
  packageExtensions,
  peerDependencyRules,
  patchedDependencies,
}
if (onlyBuiltDependencies) {
  settings['onlyBuiltDependencies'] = onlyBuiltDependencies
}
if (onlyBuiltDependenciesFile) {
  settings['onlyBuiltDependenciesFile'] = path.join(manifestDir, onlyBuiltDependenciesFile)
}
```